homeassistant/components/home_connect/api.py

```diff
@@ -348,31 +400,47 @@ class Oven(
     DeviceWithPrograms,
     DeviceWithRemoteControl,
     DeviceWithRemoteStart,
+    DeviceWithLocalControl,
+    DeviceWithProgramFinished,
 ):
     """Oven class."""
 
-    PROGRAMS = [
-        {"name": "Cooking.Oven.Program.HeatingMode.PreHeating"},
-        {"name": "Cooking.Oven.Program.HeatingMode.HotAir"},
-        {"name": "Cooking.Oven.Program.HeatingMode.TopBottomHeating"},
-        {"name": "Cooking.Oven.Program.HeatingMode.PizzaSetting"},
-        {"name": "Cooking.Oven.Program.Microwave.600Watt"},
-    ]
-
     power_off_state = BSH_POWER_STANDBY
 
+    def get_oven_temperature_sensors(self):
+        """Get a dictionary with info about oven temperature sensors."""
+        return [
+            {
+                ATTR_DEVICE: self,
+                ATTR_DESC: "Temperature",
+                ATTR_KEY: COOKING_OVEN_TEMPERATURE,
+                ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+                ATTR_UNIT: None,
```

Comment:
Shouldn't we set a temperature unit here, either Celsius or Fahrenheit?

Reply:
No.
There is no "default" unit. But we receive it in every temperature packet, and set it here: https://github.com/home-assistant/core/blob/ed2fdab090647917aef304e5ad000b4607d979ad/homeassistant/components/home_connect/sensor.py#L104

Reply:
For Celsius and Fahrenheit and other units Home Assistant has standard units that should be used. So the integration should check for a corresponding unit and then replace it with our constants, eg `TEMP_CELSIUS` and `TEMP_FAHRENHEIT`.

It's ok to not set a standard unit, but the integration should somewhere make sure that appropriate Home Assistant units are used.

Reply:
Fixed!